### ISSUE TITLE
Rabbits for dinner

### ISSUE THREAD

**Issue by digger112344** [UnofficialCrusaderPatch/UnofficialCrusaderPatch#1033]
It would be great if rabbits could be hunted as deers for meat to the granary, so it will make meat much more usable and valuable resourse

**Comment by gynt** [comment]
Duplicate of 
https://github.com/UnofficialCrusaderPatch/UnofficialCrusaderPatch/issues/168#issuecomment-453264415
https://github.com/UnofficialCrusaderPatch/UnofficialCrusaderPatch/issues/374#issuecomment-493620700
https://github.com/UnofficialCrusaderPatch/UnofficialCrusaderPatch/issues/376#issuecomment-493712843

But I will not close this issue because this is the only issue that only mentions rabbits :) so this is a good tracker issue.

**Comment by Panbutt** [comment]
I agree with the idea with hunting bunnies - rabbits reproduce *so fast* that sometimes they can straight up conquer areas of the map if you don't manually kill them with army units lol. I think hunters should prioritize deer but target rabbits if certain conditions are met -

- Switch to hunting rabbits if deer population is too low (to prevent wiping out all deer on the map, less than 6?)
- Switch to hunting rabbits if rabbit population is too high (to prevent rabbits ruining an ecosystem, more than 25?)
- If both rabbit and deer populations are too low, maybe hunters stop hunting until the populations recover? (rabbit pop less than 10?)
- The quota to return to the granary is hunting at least 5 rabbits. Each rabbit could be worth 1-2 food, returning 5-10 food at max per trip (not including fear factor). Behavior-wise, if rabbits are grouped up the hunter could kill 5 rabbits before gathering their remains. Maybe the hunting dog could get in the action too and both hunt rabbits until they meet the combined quota of 5?
- The advantage of hunting rabbits is that the hunter can clean them on the field and go to the granary immediately instead having to return to the hut first like with a deer but disadvantage is rabbits provide less food. (iirc deer provide around 15 food? maybe buff it to 20?)

The balance thought is that rabbits provide small amounts of food quickly whereas deer provide large amounts of food slowly.

Is it also possible to let deer respawn/reproduce akin to rabbits? IIRC rabbits are the only ones that can 'respawn'. It'd make hunters way more valuable and an actual tangible source of food. Obviously deer shouldn't respawn as fast as rabbits but yeah.

**Comment by gynt** [comment]
Nice contribution! Are your suggestions about single player or skirmish mode?

**Comment by GRhin** [comment]
I think the deer population conditions should be revisited.
Is less than 6 deer near keep, but 100 near enemy keep (so cant hunt them)
valid to switch? if not, you will hunt all deer you can access, while
leaving rabbits around.
Similarly rabbit population being very high but near enemy or too far away
to be practical to hunt will render hunters useless even when there is
sufficient deer nearby.
remedying these problems would be make the idea work, and i like that you
considered balance options between rabbits and deer, however I think it
would make the adjustment prohibitively complicated.
Might i suggest a simpler alternative? Have a switch in each hunters post
(similar to fletcher can switch from bows to xbows). This allows the player
to decide whether to go for rabbits or deer depending on the situation. If
the populations are too low the player can sleep the building until they
recover.
This way we are asking the player to set their own conditions instead of
asking a programmer to do it for everyone. Should theoretically be simpler
to implement and would allow the player more freedom to set their economy
how they wish. Especially when if you have different food production rates
between rabbits and deer (i would hate to not notice they switched to
rabbits and run out of food because i didnt account for it)

**Comment by Panbutt** [comment]
Having a switch between deer and rabbit is a fantastic idea, idk how that thought didn't even cross my mind lol. Yeah, giving players more control over hunting priorities is better overall from both a gameplay and coding standpoint.

**Comment by Heroesflorian** [comment]
Some more things to keep in mind: 
- Killed rabbits despawn faster than deer, and even shot deer sometimes despawn before the hunter reaches them, so a hunter hunting rabbits would rather shoot and collect them one-by-one rather than shoot multiple and try to gather what already has despawned afterwards. 
- Rabbits multiply like rabbits, but better: Even one "source rabbit" can breed up to a full group of around 32 again, and rather quickly so. 
- Repopulation of deer in SHC, especially skirmish, usually doesn't work that well, or at all, potentially making other adjustments necessary to avoid rabbits from becoming strictly better deer due to their rapid-respawning nature. For SH1, deer can repopulaty albeit more slowly and with some growing-up stages and only if a group's leading deer is still alive; when extinct, new deer spawn at a predetermined, fixed place on the map; this clearly doesn't happen for SHC skirmish though. 
- How will the AI switch between rabbits and deer?

**Comment by Panbutt** [comment]
> Some more things to keep in mind:
> 
> * Killed rabbits despawn faster than deer, and even shot deer sometimes despawn before the hunter reaches them, so a hunter hunting rabbits would rather shoot and collect them one-by-one rather than shoot multiple and try to gather what already has despawned afterwards.
> * Rabbits multiply like rabbits, but better: Even one "source rabbit" can breed up to a full group of around 32 again, and rather quickly so.
> * Repopulation of deer in SHC, especially skirmish, usually doesn't work that well, or at all, potentially making other adjustments necessary to avoid rabbits from becoming strictly better deer due to their rapid-respawning nature. For SH1, deer can repopulaty albeit more slowly and with some growing-up stages and only if a group's leading deer is still alive; when extinct, new deer spawn at a predetermined, fixed place on the map; this clearly doesn't happen for SHC skirmish though.
> * How will the AI switch between rabbits and deer?

For killed rabbits, I mentioned using the hunting dog. They move faster than the hunter so I would if it could go like this;
-Hunter kills rabbit
-Dog rushes to rabbit corpse to 'gather' it
-Dog runs back to Hunter, rinse and repeat until a number is met (around 5-6 rabbits?)

In all my time playing SHC I've never seen the deer repopulate. It might've been disabled for multiplayer/skirmish if it's there in singleplayer. Might be worth considering using the rabbit repopulation code but just slowing it down considerably.

For the AI, that's tough. I get the feeling hunting mechanics were a bit of an afterthought as there's nothing in the AICs that controls granary meat amount or amount of hunting posts. I'm unsure how to go about that.
